tests/test_optimize_shot_adapative.py

```diff
@@ -0,0 +1,522 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the shot adaptive optimizer"""
+import pytest
+from scipy.stats import multinomial
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+@pytest.fixture(autouse=True)
+def tape_mode_only():
+    """Run the test in tape mode"""
+    qml.enable_tape()
+    yield
+    qml.disable_tape()

```

Comment:
Don't forget to remove this, unless we want to do another tape-mode-sweep later. 👀

Reply:
Had the same thought as I was writing it 😬 But this is the only way to get the tests passing before #1100 is merged